server/src/main/java/org/elasticsearch/search/aggregations/InternalAggregation.java

```diff
@@ -126,23 +126,25 @@ public String getName() {
         return name;
     }
 
+    /**
+     * Creates the output from all pipeline aggs that this aggregation is associated with.  Should only
+     * be called after all aggregations have been fully reduced
+     */
+    public InternalAggregation materializePipelines(InternalAggregation reducedAggs, ReduceContext reduceContext) {
```

Comment:
I wonder if instead of adding `materializePipelines` in the `PipelineAggregator` we could keep `reduce` and just change the name here to be `reducePipeline`. The renaming is not really needed since pipeline and aggregator don't share a base class so there is no confusion ?